### ISSUE TITLE
Examine directory structure on disk to support multiple download clients

### ISSUE THREAD

**Issue by C-J1** [davestephens/ansible-nas#37]
Currently, Ansible-NAS is setup to only support torrent downloads. The defined directory structure is limited to only this configuration.

I suggest that a couple of core changes need to be made to support multiple download clients.

**Comment by C-J1** [comment]
Moved comment over from the PR:

---------------------------

A bit of work to do in here I think...

In order for all the download handlers (torrent, nzbget, sabnzbd, etc) to work smoothly with the media management tools (radarr, sonarr, etc) then the paths inside the containers need to match across the containers. With some of the tools (sonarr and radarr) you can do some remote path mapping stuff, but if we get it right at the "physical" path level, then things should just "work".

For example:

application | path on host | path in container
------------ | ------------- | -------------
transmission | /mnt/volX/downloads/transmission/complete | /downloads/transmission/complete
nzbget | /mnt/volX/downloads/nzbget/completed | /downloads/nzbget/completed
sonarr | /mnt/volX/downloads | /downloads

In the example, the idea is that sonarr (and other tools of a similar nature) will talk directly to the download clients via API and, once a download is finished, will read the path to the completed download ([ref](https://github.com/Sonarr/Sonarr/wiki/Completed-Download-Handling)). If that path doesn't match across the containers, then sonarr or radarr would not find the file. We could overcome that with remote path mappings (for apps that support it), but I feel that would add an additional level of complexity (and not work in all cases), considering that technically all of the hosts are "nas@domain" with port variations.

And this has also got me thinking longer term regarding media locations. ie/ someone mentioned airsonic... and that'd be great, but we'd then have to add a /audiobooks mount in to all the relevant download client configs. Better to just have something like a /media path (which we kind of have with "samba_shares_root"), so that again all the paths would match across all of the containers.

Regards,

CJ

PS I have a commit or two working on this. Haven't pushed it up yet, as I'm tired and am only halfway through... :)

**Comment by C-J1** [comment]
So I've got a new PR in the works for this one. Starting out with just changing how the downloads directory works. Once we get that merged in (and also so that it handles moving existing systems over to the new directory structure) we can build on it with additions of NZBGet, etc. I've already got some branches ready to go for that :)
